delira/utils/config.py

```diff
@@ -1,13 +1,509 @@
+        value : any
+            attribute value
+
+        """
+        super().__setattr__(key, self._to_config(value))
+
+    @non_string_warning
+    def __setitem__(self, key, value):
+        """
+        Set items inside dict. Supports setting of nested entries by
+        seperating the individual keys with a '.'.
+
+        Parameters
+        ----------
+        key : str
+            key for new value
+        value : any
+            new value
+        """
+        if not isinstance(key, str) or '.' not in key:
+            super().__setitem__(key, value)
+        else:
+            current_level = self
+            keys = key.split(".")
+            final_key = keys.pop(-1)
+            final_dict = self._traverse_keys(keys, create=True)
+            final_dict._set_internal_item(final_key, value)
+
+    @non_string_warning
+    def __getitem__(self, key):
+        """
+        Get single item
+
+        Parameters
+        ----------
+        key : str
+            key to desired item
+
+        Returns
+        -------
+        Any
+            value inside dict
+        """
+        if not isinstance(key, str) or '.' not in key:
+            try:
+                return super().__getitem__(int(key))
+            except (KeyError, ValueError):
+                return super().__getitem__(key)
+        else:
+            return self._traverse_keys(key.split("."), create=False)
+
+    @non_string_warning
+    def __contains__(self, key):
+        """
+        Check if key is in config
+        (also works for nested dicts with short form)
+
+        Parameters
+        ----------
+        key : str
+            key for desired value
+
+        Returns
+        -------
+        bool
+            true if key is in config
+        """
+        contain = True
+        try:
+            self[key]
+        except KeyError:
+            contain = False
+        return contain
+
+    def __str__(self):
+        """
+        String representation of config
+
+        Returns
+        -------
+        str
+            representation of config
+        """
+        return self.dumps()
+
+    def _traverse_keys(self, keys, create=False):
+        """
+        Internal helper to traverse through nested dicts
+        (iterative implementation to avoid problems with python stack)
+
+        Parameters
+        ----------
+        keys : iterable of list
+            iterable with keys which should be traversed
+        create : bool, optional
+            creates new empty configs for non existant keys, by default False
+
+        Returns
+        -------
+        Any
+            value defined by the traversed keys
+        """
+        current_level = self
+        for k in keys:
+            if k not in current_level:
+                current_level[k] = self._create_internal_dict()
+            # traverse to needed dict
+            current_level = current_level._try_key(k)
+        return current_level
+
+    def _try_key(self, key):
```

Comment:
I don't know, if we really should do it like this, I'd prefer to use strings and only raise warnings otherwise